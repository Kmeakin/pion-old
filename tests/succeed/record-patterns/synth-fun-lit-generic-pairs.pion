//~ run = "${PION} elab ${FILE}"

fun ((A, B) : (Type, Type)) ((x, y) : (A, B)) => x
