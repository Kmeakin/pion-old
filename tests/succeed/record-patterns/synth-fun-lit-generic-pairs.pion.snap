exit status: 0
stdout:
fun (a : (Type, Type)) => {
    let A : Type = a._0;
    let B : Type = a._1;
    fun (b : (a._0, a._1)) => {
        let x : a._0 = b._0;
        let y : a._1 = b._1;
        x
    }
} : fun (a : (Type, Type)) -> (a._0, a._1) -> a._0

stderr:
